addons/binding/org.openhab.binding.globalcache/src/main/java/org/openhab/binding/globalcache/handler/GlobalCacheHandler.java

```diff
@@ -788,10 +856,202 @@ private void checkConnection() {
+            serialReaderJob = null;
+            terminateSerialReader = false;
+        }
+
+        public void start() {
+            serialReaderJob = scheduledExecutorService.schedule(serialReaderRunnable, 0, TimeUnit.SECONDS);
+        }
+
+        public void stop() {
+            if (serialReaderJob != null) {
+                terminateSerialReader = true;
+                serialReaderJob.cancel(true);
+                serialReaderJob = null;
+            }
+        }
+
+        private void serialReader() {
+            logger.info("Serial reader RUNNING for {} on {}:{}", thingID(), getIP(), serialPort);
+
+            while (!terminateSerialReader) {
+                byte[] buffer;
+                try {
+                    buffer = readUntilEndOfMessage(endOfMessage);
+                    if (buffer == null) {
+                        logger.debug("Received end-of-stream from {} on {}", getIP(), serialPort);
+                        continue;
+                    }
+                    logger.debug("Rcv data from {} at {}:{}: {}", thingID(), getIP(), serialPort,
+                            getAsHexString(buffer));
+                    updateFeedbackChannel(buffer);
+                } catch (UnsupportedEncodingException e) {
+                    logger.info("Unsupported encoding exception: {}", e.getMessage(), e);
+                    continue;
+                } catch (IOException e) {
+                    logger.debug("Serial Reader got IOException: {}", e.getMessage());
+                    break;
+                } catch (InterruptedException e) {
+                    logger.debug("Serial Reader got InterruptedException: {}", e.getMessage());
+                    break;
+                }
+            }
+            logger.debug("Serial reader STOPPING for {} on {}:{}", thingID(), getIP(), serialPort);
+        }
+
+        private byte[] readUntilEndOfMessage(byte[] endOfMessageDelimiter) throws IOException, InterruptedException {
+            logger.debug("Serial reader waiting for available data");
+
+            int val;
+            ByteArrayOutputStream buf = new ByteArrayOutputStream();
+
+            // Read from the serial input stream until the endOfMessage delimiter is found
+            while (true) {
+                val = serialIn.read();
+                if (val == -1) {
+                    logger.debug("Serial reader got unexpected end of input stream");
+                    throw new IOException("Unexpected end of stream");
+                }
+
+                buf.write(val);
+                if (findEOM(buf.toByteArray(), endOfMessageDelimiter)) {
+                    // Found the end-of-message delimiter in the serial input stream
+                    break;
+                }
+            }
+            logger.debug("Serial reader returning a message");
+            return buf.toByteArray();
+        }
+
+        private boolean findEOM(byte[] buf, byte[] endOfMessage) {
```

Comment:
Ah, missed that one.  :-(  I'm gonna leave lengthEOM alone because I think it's clear given the context.